scripts/pre-commit.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/sh
```

Comment:
This is technically super fine, but I would have used a makefile (and/or a taskfile(https://taskfile.dev/)) for this.

Using a specific makefile let us improve later on and add more tasks.

Reply:
That's an idea. I always biased toward shell because it's pervasive, but I suppose that's not true on Windows, and also it isn't hard to get a hold of a make implementation. If we made a `Makefile` in the root, would people mistake that for "how the project gets built"?

Reply:
I believe that if we make proper documentation, people won't get confused.

Reply:
Makefiles let you choose what runs by default, so `make` could run a build job that just runs `cargo build`. We could then have a `make ci` job (or whatever) and `make test` so everything's bundled into one.

Reply:
In my own stuff, `make` doesn't run anything, it just prints the available actions: https://github.com/loophp/iterators/blob/main/Makefile

Reply:
this seems like it deserves its own top-level discussion. @drupol can you make an issue to continue the conversation? then we can merge this

Reply:
Sure, I can even do a P.O.C. if you want to !

Reply:
Just wanted to add here that there is a similar tool like taskfile for cargo called cargo-make https://github.com/sagiegurari/cargo-make